### PR TITLE
Update docs to add new NS, CAA and SRV records and add FAQ About who are allowed to use NS records

### PR DESCRIPTION
**Pull request by Bananalolok**
Updated the docs with the following,
      - Docs for the new records (NS, CAA, SRV)
      - Added a FAQ About who/what are allowed for NS record

### CODE REVIEW

File: src/docs/faq/index.md [hunk 1]
```diff
@@ -8,6 +8,9 @@ We support the following records:
 - MX
 - TXT
 - URL (Redirection)
+- NS (Limited Acess)
```
Comment (wdhdev) [inline src/docs/faq/index.md:11]:
```suggestion
- NS (limited access)
```

**Review by wdhdev — COMMENTED** [review]
I'll finish reviewing this this evening

File: src/docs/domain-structure/index.md [hunk 2]
```diff
@@ -144,3 +148,59 @@ Below are some examples for the given record types:
...
+    "NS": [
+      "ns1.example.com",
+      "ns2.example.com"
+    ]
+  }
+}
+```
+Note: Please refer to the [frequently asked questions](https://is-a.dev/docs/faq/) for clarification on what or who we allow NS records for.
+
+-**SRV** record: This must be a list of service records. Each record specifies the priority, weight, port, and target for a service on your domain. SRV records are often used for services such as VoIP, messaging, and more.
```
Comment (wdhdev) [inline src/docs/domain-structure/index.md:164]:
```suggestion
- **SRV** record: This must be a list of service records. Each record specifies the priority, weight, port, and target for a service on your domain. SRV records are often used for services such as VoIP, messaging, and more.
```

File: src/docs/domain-structure/index.md [hunk 3]
```diff
@@ -144,3 +148,59 @@ Below are some examples for the given record types:
...
+        "priority": 20,
+        "weight": 10,
+        "port": 9090,
+        "target": "srv2.example.com"
+      }
+    ]
+  }
+}
+```
+-**CAA** record: This must be a list of Certification Authority Authorization (CAA) records. Each record specifies the authority permitted to issue SSL certificates for your domain, with fields for `flags`, `tag`, and `value`.
```
Comment (wdhdev) [inline src/docs/domain-structure/index.md:185]:
```suggestion
- **CAA** record: This must be a list of Certification Authority Authorization (CAA) records. Each record specifies the authority permitted to issue SSL certificates for your domain, with fields for `flags`, `tag`, and `value`.
```

File: src/docs/domain-structure/index.md [hunk 4]
```diff
@@ -144,3 +148,59 @@ Below are some examples for the given record types:
...
+        "tag": "issuewild",
+        "value": "comodoca.com"
+      }
+    ]
+  }
+}
+```
+
+### proxy (optional)
+This feild is where you specify whether you want to enable Cloudflare proxy for your domain.
```
Comment (wdhdev) [inline src/docs/domain-structure/index.md:206]:
```suggestion
### proxied (*optional*)
Enable Cloudflare proxy for your domain. Disabled by default.
```

File: src/docs/faq/index.md [hunk 5]
```diff
@@ -8,6 +8,9 @@ We support the following records:
 - MX
 - TXT
 - URL (Redirection)
```
Comment (wdhdev) [inline src/docs/faq/index.md:10]:
```suggestion
- URL (redirection)
```

File: src/docs/faq/index.md [hunk 6]
```diff
@@ -45,6 +48,18 @@ Please open a help thread via any of our help channels and we will look into it.
 ## Can I use my domain for a minecraft server?
 Sure, but it needs to be a nested subdomain (mc.example.is-a.dev). You would normally use a A record for this.
 
+## Can I Use NS Records?
```
Comment (wdhdev) [inline src/docs/faq/index.md:51]:
```suggestion
## Can I use NS records?
```

File: src/docs/faq/index.md [hunk 7]
```diff
@@ -45,6 +48,18 @@ Please open a help thread via any of our help channels and we will look into it.
...
+We allow NS records for the following reasons and users:
+  - is-a.dev maintainers
+  - Users who are using **multiple** Cloudflare (or similar providers') services, which often require their own zones (e.g., DynDNS, cloudflared tunnels, etc.)
+  - Users who require a private zone, as they are using their home IP addresses behind Cloudflare (These cases will be carefully reviewed, as we understand some may claim to use a home IP address when they are not.)
+  - Users with a lot of nested subdomains *may* be allowed to register NS records so they do not have to manage multiple files.
+  - Aternos users (specifically Aternos, as they require NS records and you cannot modify DNS records under them).
+  
+Note: These guidelines are not set in stone and may change. You are not limited to fitting only one of the criteria above; use them as a general guide. I will personally review each NS record request.
```
Comment (wdhdev) [inline src/docs/faq/index.md:59]:
```suggestion
Note: These guidelines are not set in stone and may change. You are not limited to fitting only one of the criteria above; use them as a general guide. [@wdhdev](https://github.com/wdhdev) personally review each NS record request.
```

**Review by wdhdev — CHANGES_REQUESTED** [review]

**Review by orxngc — APPROVED** [review]

**Comment by orxngc** [comment]
this seems fine to me, i'll merge this

**Pull request merged**